plugins/core/hu.bme.mit.gamma.ui/src/hu/bme/mit/gamma/ui/GammaApi.java

```diff
@@ -179,6 +180,8 @@ else if (task instanceof TestGeneration) {
 									TestGeneration testGeneration = (TestGeneration) task;
 									TestGenerationHandler handler = new TestGenerationHandler(file);
 									handler.execute(testGeneration, projectName);
+									MakefileGenerator.tests.clear(); // Manual reset
```

Comment:
This is not a good pattern. Make sure to include the relevant C-related constructs only in the corresponding methods. Maybe the `MakefileGenerator` should be an instantiable class and then, these clears would not need to be carried out.